scenes/enemies/guard/guard_movement.gd

```diff
@@ -0,0 +1,89 @@
+@tool
+class_name GuardMovement
+extends Node2D
+
+## Emitted when [member still_time_left] reaches 0
+signal still_time_finished
+## Emitted when [member guard] reached [member destination]
+signal destination_reached
+## Emitted when [member guard] got stuck trying to reach [member destination]
+signal path_blocked
+
+@export var guard: Guard
```

Comment:
If we can safely assume that the owner of this node is a Guard, then this can be simply:
```suggestion
@onready var guard: Guard = owner
```
And the `_ready()` below can be removed. I'm switching to this in some nodes of Player in my branch.

Reply:
yeah, I think that for a node as tighly coupled as this one that would be fine 👍🏾